backend/storage-service/src/jvmMain/kotlin/file/http/FileSynchronizationController.kt

```diff
@@ -0,0 +1,43 @@
+package dk.sdu.cloud.file.http
+
+import dk.sdu.cloud.calls.server.RpcServer
+import dk.sdu.cloud.service.Controller
+import dk.sdu.cloud.Actor
+import dk.sdu.cloud.calls.server.securityPrincipal
+import dk.sdu.cloud.file.api.FileSynchronizationDescriptions
+import dk.sdu.cloud.file.services.SynchronizationService
+
+class SynchronizationController(
+    private val synchronizationService: SynchronizationService
+) : Controller {
+    override fun configure(rpcServer: RpcServer) = with(rpcServer) {
+        implement(FileSynchronizationDescriptions.retrieveFolder) {
+            ok(synchronizationService.retrieveFolder(Actor.SystemOnBehalfOfUser(ctx.securityPrincipal.username), request.path))
```

Comment:
```suggestion
            ok(synchronizationService.retrieveFolder(actorAndProject.actor, request.path))
```

This probably requires an import. I am pretty sure this functionality is present on `master`. If not, we can simply port it back.